src/ucs/stats/stats.c

```diff
@@ -460,6 +477,184 @@ void ucs_stats_node_free(ucs_stats_node_t *node)
+                (ucs_stats_context.aggrgt_cnt_names_db_size *
+                 sizeof(*ucs_stats_context.aggrgt_cnt_names_db));
+        ucs_stats_context.aggrgt_cnt_names_db = ucs_realloc(
+                ucs_stats_context.aggrgt_cnt_names_db, total_alloc_size_bytes,
+                (const char*)UCS_STATS_AGGRGT_SUM_CNT_NAMES_DB);
+
+        cnt_agrgt_sum_index = offset;
+        ucs_for_each_bit(counter_index, filter_node->counters_bitmask) {
+            ucs_list_for_each(temp_node, &filter_node->type_list_head,
+                              type_list) {
+                ucs_stats_context.aggrgt_cnt_names_db[cnt_agrgt_sum_index].name =
+                        node->cls->counter_names[counter_index];
+                ucs_stats_context.aggrgt_cnt_names_db[cnt_agrgt_sum_index]
+                        .class_name = node->cls->name;
+            }
+            cnt_agrgt_sum_index++;
+        }
+
+        ucs_stats_context.aggrgt_sum_db_size++;
+    }
+
+    pthread_mutex_unlock(&ucs_stats_context.lock);
+
+    return offset;
+}
+
+
+/**
+ * Returns the aggregate-sum of all counters from all interfaces for a compact statistics trace.
+ * Used for Score-P UCX plugin to collect a compact trace of the counters.
+ *
+ * @param [in]  node                      Pointer to the statistics node.
+ * @param [in]  sel                       Whether or not to recursively iterate through the children.
+ * @param [in]  current_num_counters      Current number of counters iterated through (call with 0).
+ * @param [out] ucs_stats_aggregate_sum   The aggregate-sum of all counters of each class/type.
+ * @param [in]  num_stats_counters_max The number of elements in ucs_stats_aggregate_sum.
+ *
+ * Complexity: O(n)
+ *
+ * Returns: The total number of counters.
+ *
+ * Remarks: The function is recursive.
+ */
+static size_t ucs_stats_aggregate_sum_recurs(ucs_stats_node_t *node,
+                                             ucs_stats_children_sel_t sel,
+                                             size_t current_num_counters,
+                                             ucs_stats_counter_t *aggregate_sum,
+                                             size_t num_stats_counters_max)
+{
+    ucs_stats_filter_node_t *filter_node = node->filter_node;
+    ucs_stats_node_t *temp_node;
+    ucs_stats_node_t *child;
+    uint32_t counter_index;
+    size_t n_counters = current_num_counters;
+    uint32_t cnt_agrgt_sum_index;
+    uint32_t counter_output_index;
+    uint32_t num_counters_filtered;
+
+    /* Filter output */
+    counter_output_index  = 0;
+    num_counters_filtered = ucs_popcount(filter_node->counters_bitmask);
+    cnt_agrgt_sum_index   = 0;
+    ucs_for_each_bit(counter_index, filter_node->counters_bitmask) {
+        ucs_list_for_each(temp_node, &filter_node->type_list_head, type_list) {
+            cnt_agrgt_sum_index = counter_output_index +
+                                  ucs_stats_clsid_to_agrgt_sum_base_offset_get(
```

Comment:
pls make better and shorter name for ucs_stats_clsid_to_agrgt_sum_base_offset_get